client/src/screens/availability/Index.js

```diff
@@ -52,27 +52,74 @@ class Index extends Component {
     this.props.dispatch(setSelectedDate(dt.unix()));
   };
 
-  clearItems = () => {
-    this.props.dispatch(clearAvailability());
-  };
-
   render() {
+    const { loading, user } = this.props;
+
+    if (loading || !user) {
+      return (
+        <Container>
+          <Progress />
+        </Container>
+      );
+    }
+
+    // are they part of any groups?
+    if (user.groups.length === 0) {
+      return (
+        <Container>
+          <Center>
+            <Text>You are not part of any groups.</Text>
+          </Center>
+        </Container>
+      );
+    }
+
+    // do any of the groups have schedules?
+    // expands to check date range of requests
+    if (user.schedules.length === 0) {
+      return (
+        <Container>
+          <Center>
+            <Text>None of the groups you belong to have any open requests.</Text>
+          </Center>
+        </Container>
+      );
+    }
+
+    // move to redux filters alter
     const momentDate = moment.unix(this.props.selectedDate);
 
     // get start of week unix timestamp
     const startOfWeek = momentDate
       .clone()
       .isoWeekday(1)
-      .startOf('isoweek');
+      .startOf('isoweek')
+      .unix();
+
     const endOfWeek = momentDate
       .clone()
       .isoWeekday(1)
-      .endOf('isoweek');
+      .endOf('isoweek')
+      .unix();
 
-    // will be replaced by graphql calls
-    const filteredItems = this.props.items.filter(
-      item => item.startDateTime >= startOfWeek.unix() && item.startDateTime < endOfWeek.unix(),
-    );
+    const filteredItems = [];
+    _.forEach(user.schedules, (schedule) => {
```

Comment:
nit: I don't think you need `lodash` for this anymore - you can just use `user.schedules.forEach()`. I could be mistaken though.

Reply:
Yes - I think you are right (future pull request to remove).